packages/synapse-react-client/src/utils/functions/queryUtils.ts

```diff
@@ -241,3 +245,40 @@ export function removeEmptyQueryParams(q: Query) {
 
   return query
 }
+
+/**
+ * Given a FacetColumnResult and a set of ColumnModels, return the ColumnModel that
+ * matches the FacetColumnResult.
+ * @param facet
+ * @param columnModels
+ */
+export function getCorrespondingColumnForFacet(
+  facet: FacetColumnResult,
+  columnModels: ColumnModel[],
+): ColumnModel | JsonSubColumnModel | undefined {
+  let columnModel: ColumnModel | JsonSubColumnModel | undefined =
+    columnModels.find(model => model.name === facet.columnName)
+  if (facet.jsonPath && columnModel && columnModel.jsonSubColumns) {
+    columnModel = columnModel.jsonSubColumns.find(
+      cm => cm.jsonPath === facet.jsonPath,
+    )
+  }
+  return columnModel
+}
+
+/**
+ * Given a set of FacetColumnRequests, return the FacetColumnRequest that matches the given UniqueFacetIdentifier.
+ * @param facetDefinition
+ * @param selectedFacets
+ */
+export function getCorrespondingSelectedFacet(
+  facetDefinition: UniqueFacetIdentifier,
+  selectedFacets?: FacetColumnRequest[],
+): FacetColumnRequest | undefined {
+  return selectedFacets?.find(selectedFacet =>
+    isEqual(
+      pick(selectedFacet, ['columnName', 'jsonPath']),
+      pick(facetDefinition, ['columnName', 'jsonPath']),
+    ),
+  )
+}
```

Comment:
Since getting the facet now requires `columnName` and `jsonPath`, pull this into a utility. A later PR will actually use this.